src/main/java/com/minecolonies/coremod/colony/buildings/BuildingFisherman.java

```diff
@@ -39,6 +41,8 @@
     public BuildingFisherman(final Colony c, final BlockPos l)
     {
         super(c, l);
+
+        keepX.put(new ItemStorage(Items.FISHING_ROD, 0, 0, true), 1);
```

Comment:
Reminder for me to create an issue for improving the performance of this on a later date. This is much better than before though